File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    
    var chars = ["A","E","I","O","U"]
    var cases = []
    const make_cases = (count, cur_string) => {
        
        if (cur_string != "") {
            cases.push(cur_string)
        }

        if (count == 5) {
            return
        }
        
        for (let i = 0; i < chars.length; i++) {
            make_cases(count + 1, cur_string + chars[i])
        }
        
    }
    make_cases(0,"")
    
    cases.sort()
    
    answer = cases.indexOf(word) + 1
    
    
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

void make_cases(int count, string cur_word, vector<string>& cases) {
    vector<char> alphabets = {'A', 'E', 'I', 'O', 'U'};
    
    if (count > 0) {
        cases.push_back(cur_word);
    }
    
    if (count == 5) {
        return;
    }
    
    for (int i = 0; i < 5; i++) {
        make_cases(count + 1, cur_word + alphabets[i], cases);
    }
    
}


int solution(string word) {
    int answer = 0;
    
    vector<string> cases;
    
    make_cases(0, "", cases);
    
    sort(cases.begin(), cases.end());
    
    for (int i = 0; i < cases.size(); i++) {
        if (cases[i] == word) {
            return i + 1;
        }
    }
    
    
    
    return answer;
}